Added week3 homework files

jaimakena opened the pull request:
Javascript1-Week3 Homework

jaimakena committed:
Added week3 homework files

obelitz commented on javascript/javascript1/week3/itemArrayRemoval.js:
```diff
@@ -0,0 +1,26 @@
+  const nameToRemove = "Ahmad";
+  //To remove an item using indexOf and position if we do not know the index:
+
+  let position = names.indexOf(nameToRemove);
+  let removeditem = names.splice(position, 1);
```
These should be `const`.

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/noteTakingApp.js:
```diff
@@ -0,0 +1,37 @@
+const notes = [];
+
+function saveNote(content, id){
+   return notes.push({content: content, id: id});
```
There's a shorthand you can use for objects, when the propert and variable name is the same:
```javascript
{ content, id } // is the same as { content: content, id: id }
```

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/noteTakingApp.js:
```diff
@@ -0,0 +1,37 @@
+      return notes[i];
+    }
+  }
+    if(!id){
```
What if it's there, but it's not a number?

jaimakena replied:
I have changed it to check for NAN

obelitz commented on javascript/javascript1/week3/seriesDuration.js:
```diff
@@ -0,0 +1,38 @@
+    let totalTime = 0;
+    for(let i = 0; i < seriesDurations.length; i++)
+    {
+    let daysSeries = seriesDurations[i].days + seriesDurations[i].hours/24 + seriesDurations[i].minutes/1440;
+    let percentOfSeries = daysSeries/daysOfAvgLife * 100; 
```
These should be `const`.

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/seriesDuration.js:
```diff
@@ -0,0 +1,38 @@
+    let daysSeries = seriesDurations[i].days + seriesDurations[i].hours/24 + seriesDurations[i].minutes/1440;
+    let percentOfSeries = daysSeries/daysOfAvgLife * 100; 
+    console.log(`${seriesDurations[i].title} took ${percentOfSeries.toFixed(2)}% of my life.`);
+    totalTime = totalTime + percentOfSeries;
```
You can use the shorthand `+=` for this:
`totalTime += percentOfSeries`

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/smartPhoneUsageApp.js:
```diff
@@ -0,0 +1,42 @@
+//Smart-Phone Usage app:
+//Adding an activity
+let activities = [];
```
Arrays can usually be `const`.

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/smartPhoneUsageApp.js:
```diff
@@ -0,0 +1,42 @@
+let activities = [];
+const limit = 110;
+
+function addActivity(date, activity, duration)
+{
+  if(!date){
+    let todayDate = new Date();
+    date = todayDate.toLocaleDateString();
+  }
```
Nice! 
`todayDate` be a `const` though.

You could also make use of "default parameters":
```javascript
function addActivity(
  date = new Date().toLocaleDateString, 
  activity, 
  duration
) {
```

This will set `date` to `new Date().toLocaleDateString()` if no date is provided.

jaimakena replied:
Done

obelitz commented on javascript/javascript1/week3/smartPhoneUsageApp.js:
```diff
@@ -0,0 +1,42 @@
+function showStatus(activities){
+  let sumOfDuration = 0;
+  if(activities.length === 0){
+    console.log('Add some activities before calling showStatus');
```
You should remember to return here, so the function doesn't keep running.

jaimakena replied:
Yes-Done

obelitz commented on javascript/javascript1/week3/smartPhoneUsageApp.js:
```diff
@@ -0,0 +1,42 @@
+function addActivitySmart(activity, duration)
+{
+  let todayDate = new Date();
+  date = todayDate.toLocaleDateString();
```
`date` has no variable keyword.
`const`

jaimakena replied:
I forgot to add the keyword. Now added it

obelitz commented on javascript/javascript1/week3/timeTravel.js:
```diff
@@ -0,0 +1,16 @@
+  
+  function getTravelTime(travelInformation){
+    
+    return Math.trunc(travelInformation.destinationDistance * 60/travelInformation.speed);
```
Never seen this one before. Nice!

obelitz commented on javascript/javascript1/week3/timeTravel.js:
```diff
@@ -0,0 +1,16 @@
+    return Math.trunc(travelInformation.destinationDistance * 60/travelInformation.speed);
+  }
+
+  const travelTime = getTravelTime(travelInformation);
+  const timeInHours = Math.floor(travelTime/60);
+  const timeInMinutes = travelTime % 60;
```
These calculations should be inside the function, so all it returns is 
```javascript
`Travel time: ${timeInHours} hours and ${timeInMinutes} minutes`
```

jaimakena replied:
Yeah. I did not check that before- I changed it now

obelitz reviewed (commented):
Good job!

Check your travel time assignment. 
The function seems to be returning something else than expected

jaimakena committed:
added changes from review

jaimakena commented:
> Good job!
> 
> Check your travel time assignment.
> The function seems to be returning something else than expected

Yes-I have checked and I have changed that. 
I have also added the optional homework for the smartPhoneUsage app- Could you please review that as well.
Thanks

obelitz commented:
Hi Jai
I missed your comment about smartPhoneUsage app :) 
It looks good!
